airflow/models/dag_version.py

```diff
@@ -0,0 +1,169 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+import random
+import string
+from typing import TYPE_CHECKING
+
+import uuid6
+from sqlalchemy import Column, ForeignKey, Integer, UniqueConstraint, func, select
+from sqlalchemy.orm import relationship
+from sqlalchemy_utils import UUIDType
+
+from airflow.models.base import Base, StringID
+from airflow.utils import timezone
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.utils.sqlalchemy import UtcDateTime, with_row_locks
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+
+log = logging.getLogger(__name__)
+
+
+def _gen_random_str():
+    return "".join(random.choices(string.ascii_letters + string.digits, k=10))
+
+
+class DagVersion(Base):
+    """Model to track the versions of DAGs in the database."""
+
+    __tablename__ = "dag_version"
+    id = Column(UUIDType(binary=False), primary_key=True, default=uuid6.uuid7)
```

Comment:
why binary false?

Reply:
By default, sqlalchemy-utils UUIDType uses binary for DBs that don't support UUID e.g sqlite. With binary False, it will use Char 32 to store the generated UUID: https://github.com/kvesteri/sqlalchemy-utils/blob/baf53cd1a3e779fc127010543fed53cf4a97fe16/sqlalchemy_utils/types/uuid.py#L40

Reply:
Paging @kaxil - I know you've been looking at UUID stuff recently. Any input/thoughts here?

Reply:
added comment here: https://github.com/apache/airflow/pull/42913/files#r1823527044

tldr: Only Postgres supports native UUIDs. If we want to use UUID7, we might not be able to use sqlalchemy-utils (or even need it) as it uses native UUID from python stdlib that uses UUID4

Reply:
IIUC this parameter does not matter on a database that _does_ support native UUID. The question here therefore is whether we want to store the UUID as a string or binary blob on non-Postgres.

Reply:
>The question here therefore is whether we want to store the UUID as a string or binary blob on non-Postgres.

Store it as string for non-Postgres dbs -- which @ephraimbuddy has already done with `binary=False` arg here.

@ephraimbuddy : You should add `sqlalchemy-utils` to `hatch_build.py`

Reply:
Ok. Thanks